app/models/review.rb

```diff
@@ -1,10 +1,21 @@
 class Review < ActiveRecord::Base
   belongs_to :user
   belongs_to :location
+  has_many :votes
 
   validates :rating, presence: true, inclusion: { in: 1..5 }
   validates :user, presence: true
   validates :location, presence: true
 
+  def user_vote(user)
+    vote = votes.find_by(user: user)
+    vote.id
+  end
+
+  def vote_value(user)
```

Comment:
DRY this up a bit. I think one method here should suffice as opposed to two (find user perhaps?) Also, you're not passing in a user for line 15 as an argument, you're passing in a user ID.